treeherder/perf/email.py

```diff
@@ -0,0 +1,222 @@
+     It does this every hour, as long as it doesn't exceed the daily limit.
+    > __Here's a summary of latest backfills:__
+    ---
+        """
+
+    TABLE_HEADERS = """
+    | Alert summary | Alert | Job symbol | Total backfills (aprox.) | Push range |
+    | :---: | :---: | :---: | :---: | :---: |
+        """
+
+    def __init__(self):
+        self._raw_content = None
+
+    def include_records(self, records: List[BackfillRecord]):
+        self._initialize_report_intro()
+
+        for record in records:
+            self._include_in_report(record)
+
+    def _initialize_report_intro(self):
+        if self._raw_content is None:
+            self._raw_content = self.DESCRIPTION + self.TABLE_HEADERS
+
+    def _include_in_report(self, record: BackfillRecord):
+        new_table_row = self._build_table_row(record)
+        self._raw_content += f"{new_table_row}\n"
+
+    def _build_table_row(self, record: BackfillRecord) -> str:
+        alert_summary = record.alert.summary
+        alert = record.alert
+        job_type = record.job_type
+        total_backfills = record.total_backfills_triggered
+        push_range = self.__build_push_range(record.get_context())
+
+        # some fields require adjustments
+        summary_id = alert_summary.id
+        alert_id = alert.id
+        job_symbol = str(job_type)
+
+        return f"| {summary_id} | {alert_id} | {job_symbol} | {total_backfills} | {push_range} |"
+
+    def __build_push_range(self, backfill_context: List[dict]) -> str:
+        """
+        Provides link to Treeherder' s Job view
+        """
+        from_push, to_push = self.__fetch_border_pushes(backfill_context)  # suspect range as tuple
+
+        repository = from_push.repository.name
+        from_change = from_push.revision
+        to_change = to_push.revision
+
+        return f"{settings.SITE_URL}/jobs?repo={repository}&fromchange={from_change}&tochange={to_change}"
+
+    def __fetch_border_pushes(self, backfill_context) -> Tuple[Push, Push]:
+        from_datapoint = backfill_context[0]
+        to_datapoint = backfill_context[-1]
+
+        from_push = from_datapoint['push_id']
+        to_push = to_datapoint['push_id']
+
+        from_push = Push.objects.get(id=from_push)
+        to_push = Push.objects.get(id=to_push)
+        return from_push, to_push
+
+    def __str__(self):
+        if self._raw_content is None:
+            raise ValueError("No content set")
+        return self._raw_content
+
+
+class BackfillNotificationWriter(EmailWriter):
+    def _write_address(self):
+        self._email.address = FXPERF_TEST_ENG_EMAIL
+
+    def _write_subject(self):
+        self._email.subject = "Automatic Backfilling Report"
+
+    def _write_content(self, must_mention: List[BackfillRecord]):
+        content = BackfillReportContent()
+        content.include_records(must_mention)
+
+        self._email.content = str(content)
+
+
+# For performance data cycling
+class DeleteReportContent:
```

Comment:
I feel like `DeletionReportContent` is a better name. What do you think? `Delete` sounds like you remove the report content.

Reply:
I agree: we should rename this class.